### ISSUE TITLE
What am I doing wrong?

### ISSUE THREAD

**Issue by joelnewcomer** [phonegap/phonegap-plugin-barcodescanner#230]
I have spent about 6 hours reading support topics and I've tried a dozen things but nothing is working. I have created a button and when it is clicked, it does nothing. I am using the PhoneGap mobile app on an iPhone 6 with iOS 9.3.1 for testing. I have run "cordova platform add ios" successfully and "cordova build" is also successful. I installed the plugin using "phonegap plugin add phonegap-plugin-barcodescanner" I have the following line in my config.xml: 
```
<plugin name="phonegap-plugin-barcodescanner" spec="3.1.0"/> 
```
My HTML is this:
```html
<a target="_blank" href="javascript:scan();" style="text-decoration: none"><button>Scan</button></a>
```
and my javascript is:
```js
			$(document).on('deviceready', function() {
				function scan() {
					cordova.plugins.barcodeScanner.scan(
					   function (result) {
					       alert("We got a barcode\n" +
					             "Result: " + result.text + "\n" +
					             "Format: " + result.format + "\n" +
					             "Cancelled: " + result.cancelled);
					   }, 
					   function (error) {
					       alert("Scanning failed: " + error);
					   },
					   {
					       "preferFrontCamera" : true, // iOS and Android
					       "showFlipCameraButton" : true, // iOS and Android
					       "prompt" : "Place a barcode inside the scan area", // supported on Android only
					       "formats" : "QR_CODE,PDF_417", // default: all but PDF_417 and RSS_EXPANDED
					       "orientation" : "landscape" // Android only (portrait|landscape), default unset so it rotates with the device
					   }
					);
				}
			});		
```
When I click 'Scan' in the PhoneGap app, nothing happens. What am I doing wrong? Am I missing something?

**Comment by EddyVerbruggen** [comment]
- Are you running this code inside the phonegap developer app? If so try deploying it as a standalone app.
- Don't wrap the `scan` function in a `deviceready` handler if you're triggering it from a button anyway
- Is the CSP meta tag disabled or changed from the default? If not inline JS is disabled
- Is there a JS error on the page? You can catch that by adding this to the `<head>`:

```js
window.onerror = function(error, file, line) {
  alert(error + ", " + file + ", " + line);
}
```

**Comment by joelnewcomer** [comment]
Wow! Eddy, you are awesome! Thank you for the quick response and the very thorough response. I was hoping to avoid deploying it as an app because this is an experiment and we didn't want to fork out the money for the developer's license, but if we need to do it, we'll do it. I moved the JS to not be inline because I assume it's best practice not to make it inline. Thank you so much for the solution to catch JS errors. I did that and I got an error indicating that cordova.plugins.barcodeScanner is undefined. I just now found someone who said that "the PhoneGap Developer app does not support third-party plugins at this time." So I believe that is the problem. Thank you again for the very helpful response.
